### DISCUSSION TITLE
How do you start the software?

### DISCUSSION THREAD

**Discussion by matthandford** [gusmanb/logicanalyzer#23] [Q&A] [answered]
This looks like a really great way to test out a logic analyzer as a tool without spending a bunch of money but I am having trouble getting started. I have been looking through the code base and I have no idea how to open the software. I don't see any executables or starting points. I downloaded visual studio to open the c# sln files but only get errors when I try to run it.  The readme says the software is self explanatory but I can't figure out how to start it. Am I supposed to go through cmd line? Git bash? I am using windows 10.

**Comment by matthandford** [comment, marked as answer]
Never mind. I figured it out. Every time I get frustrated enough to ask for help, I figure it out 2 seconds after I post. I was downloading the source code instead of the releases (located to the right hand side of the main page).

**Comment by Gustavomurta** [comment]
For those who want to compile the solution and don't know Visual Studio and C# well - like me.

Download and install **Visual Studio - Community version 2022**. You will need a Microsoft user to activate it.
https://learn.microsoft.com/visualstudio/install/install-visual-studio?view=vs-2022
Install this package - **Universal Windows Platform Development.** 

On the main project page at Github, click the green button and download the ZIP. 
Unzip this folder and open it in **Visual Studio 2022**.
The **Software** folder contains the complete project solution to be compiled.
In the Visual Studio - **Solution Manager tab**, find the file - **LogicAnalyzer.sln**
Compile the entire project by right-clicking and selecting - compile.

After compiling the entire solution, in this folder:
**logicanalyzer-master\Software\LogicAnalyzer\LogicAnalyzer\bin\Debug\net6.0**
You can find this file = **LogicAnalyzer.exe**  (execute it for run the program).

**Comment by OslayViera** [comment]
For some reason, I am unable to start the software. I try to open it as it should be, CLCapture.exe, but all I get is a console that lasts for a second and then nothing. I thought maybe the Pico should be connected to the PC, but it remains the same. I have the Frameworks installed, I tried running it as an Administrator, but I'm still at the same point where I started. I need help

**Reply by gusmanb** [reply]
CLCapture is the "command line capture", if you want the GUI app then you must use the "LogicAnalyzer" package.

You have all the information in [the wiki](https://github.com/gusmanb/logicanalyzer/wiki/05---The-CLCapture-program)

**Comment by ranranff** [comment]
This problem initially troubled me as well, because I was not familiar with .NET. However, I later found a clumsy method to run it on macOS.

1、cd logicanalyzer/Software/LogicAnalyzer
2、dotnet restore
3、cd logicanalyzer/Software/LogicAnalyzer/LogicAnalyzer
4、dotnet run

**Reply by gusmanb** [reply]
You don't need to install .net framework nor build it on the machine (unless you want to). If you download the pre-built package you can follow the instructions [in the wiki](https://github.com/gusmanb/logicanalyzer/wiki/04---LogicAnalyzer-Software#notes-for-linux-and-macos-users) to remove the files from quarantine and be able to run it.

Cheers.
